packages/avatar/tests/avatar.test.tsx

```diff
@@ -1,7 +1,41 @@
 import * as React from "react"
-import { render, testA11y } from "@chakra-ui/test-utils"
+import { render, testA11y, waitFor } from "@chakra-ui/test-utils"
 import { Avatar, AvatarBadge } from "../src"
 
+const DELAY = 0
+const LOAD_IMAGE = "load"
+const ERROR_IMAGE = "error"
+const orignalImage = window.Image
+
+const mockImage = (loadState: string) => {
+  jest.useFakeTimers()
+  ;(window.Image as unknown) = class MockImage {
+    onload: () => void = () => {}
+    onerror: () => void = () => {}
+    src: string = ""
+    constructor() {
+      if (loadState === LOAD_IMAGE) {
+        setTimeout(() => {
+          this.onload()
+        }, DELAY)
+      }
+      if (loadState === ERROR_IMAGE) {
+        setTimeout(() => {
+          this.onerror()
+        }, DELAY)
+      }
```

Comment:
We could reduce the duplicate lines with a switch statement:

```suggestion
      setTimeout(() => {
        switch (loadState) {
          case LOAD_IMAGE:
            this.onload()
            break
          case ERROR_IMAGE:
            this.onerror()
            break
          default:
            return
        }
      }, DELAY)
```